lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -746,6 +746,33 @@ class TFFunctionPartition {
 };
 } // end anonymous namespace
 
+
+/// Determine the marking strategy (Mark::Copy / Mark::Send) according
+/// to the type of TermInst of a predecessor BasicBlock.
+/// For general terminators (like br/cond_br), we'd generate code to
+/// run on accelerator;
+/// for other SIL-specific terminators (like try_apply), we'd keep it
+/// running on the host and send its result to accelerator.
+//@{
+static inline bool shouldMarkCopy(TermInst *predTerm) {
+  auto termKind = predTerm->getTermKind();
+  return termKind == TermKind::BranchInst ||
+      termKind == TermKind::CondBranchInst;
+}
+static inline bool shouldMarkSend(TermInst *predTerm) {
+  // TODO : support SwitchValueInst and CheckedCastValueBranchInst
```

Comment:
If you already have test cases for them, please feel free to add them as commented out to the test file(s).

Reply:
I have troubles writing test cases for both. So, I'd suggest adding them once the test is done.

`CheckedCastValueBranchInst` can be achieve by adding `-enable-sil-opaque-values` option, however it crashes TFDeabstraction pass.

`SwitchValueInst` requires `-Onone` to stop being transformed into `cond_br`, but `-Onone` crashes SILVerifier in TFPartition on the code below:
```swift
import Tensorflow
public func test(_ x: Bool) {
  var b = Tensor<Float>(2.0)
  switch (x) {
    case true: b += 1.0
    default: b += 2.0
  }
  b -= 1.0
}
```

Reply:
Anyway, I just have them included in the test file as commented out.

Reply:
Thanks. This is sufficient for now. we'll follow up on the -Onone issue when deasbtraction is done.